### PR TITLE
dbt 0.5.4 (new formula)

### PR DESCRIPTION
**Pull request by cmcarthur**
Adds a new formula for dbt (data build tool).

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

`brew audit --new-formula dbt` returns two problems:

```
dbt:
  * Stable: version 0.5.4 is redundant with version scanned from URL
  * GitHub repository not notable enough (<20 forks, <20 watchers and <50 stars)
Error: 2 problems in 1 formula
```

The version is taken straight from a tag on Github. To me, that seems like the ideal URL / version combo, let me know if / how you'd like me to resolve that.

On the second issue, this is mostly used by semi-technical users so Github forks / stars are not their go-to engagement mechanism :) Please let me know if that's a dealbreaker.

### CODE REVIEW

**Comment by woodruffw** [comment]
We use forks/stars to get a good feel for how popular a potential package is. More specialized programs tend to collect dust and leave us with a greater maintenance workload despite being downloaded relatively rarely. We then end up periodically removing unmaintained/low-frequency packages and ruining those rare users' days.

IMO we should pass on this in favor of upstream hosting their own tap, at least until it becomes more popular. I'll leave some comments, in either case.

File: Formula/dbt.rb [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    sha256 "0ff87dd5d50425084f7219cb6f86bb3eb5aa29063f53d50bf270ed007e941069"
+  end
+
+  def install
+    virtualenv_install_with_resources
+    bin.install_symlink "#{libexec}/bin/dbt" => "dbt"
+  end
+
+  test do
+    system "#{bin}/dbt", "help"
```
Comment (woodruffw) [inline Formula/dbt.rb]:
```
==> /usr/local/Cellar/dbt/0.5.4/bin/dbt help
usage: dbt: data build tool [-h] [--version]
                            {init,clean,compile,debug,deps,archive,run,seed,test}
                            ...
dbt: data build tool: error: invalid choice: 'help' (choose from 'init', 'clean', 'compile', 'debug', 'deps', 'archive', 'run', 'seed', 'test')
Error: dbt: failed
```

**Comment by cmcarthur** [comment]
@woodruffw thanks for the comments. I've pushed commits to fix these issues, let me know if you'd like me to amend into one commit and so on.

**Comment by woodruffw** [comment]
I think you need to rebase -- it looks like you pulled some history onto this branch by accident.

**Comment by cmcarthur** [comment]
@woodruffw done

File: Formula/dbt.rb [hunk 2]
```diff
@@ -0,0 +1,161 @@
+class Dbt < Formula
+  include Language::Python::Virtualenv
+
+  desc "Data build tool"
+  homepage "https://github.com/analyst-collective/dbt"
+  url "https://github.com/analyst-collective/dbt/archive/v0.5.4.tar.gz"
+  sha256 "df830bb0355e46608727fc6c59fa4aadd2514aec227e55765fe56abcfc72918c"
+
+  depends_on :python3
```
Comment (MikeMcQuaid) [inline Formula/dbt.rb]: Should perhaps be `"python3"`; worth testing locally to see if that works.

File: Formula/dbt.rb [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  depends_on "openssl"
+  depends_on :postgresql
```
Comment (MikeMcQuaid) [inline Formula/dbt.rb]: Should be `"postgresql"`

File: Formula/dbt.rb [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    sha256 "0ff87dd5d50425084f7219cb6f86bb3eb5aa29063f53d50bf270ed007e941069"
+  end
+
+  def install
+    virtualenv_install_with_resources
+    bin.install_symlink "#{libexec}/bin/dbt" => "dbt"
+  end
+
+  test do
+    system "#{bin}/dbt", "-h"
```
Comment (MikeMcQuaid) [inline Formula/dbt.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by cmcarthur** [comment]
@MikeMcQuaid I stringified the `depends_on` entries, and added a more sophisticated test. Tested with ```
brew uninstall dbt
brew install --build-from-source dbt
brew test dbt
```

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**